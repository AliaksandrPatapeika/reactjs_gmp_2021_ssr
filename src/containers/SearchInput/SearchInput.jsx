import './SearchInput.less';

import PropTypes from 'prop-types';
import React from 'react';

const SearchInput = ({value, onChange}) => (
  <input
    className="searchInput"
    placeholder="What do you want to watch?"
    autoComplete="off"
    value={value}
    onChange={onChange}
  />
);

SearchInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default SearchInput;
