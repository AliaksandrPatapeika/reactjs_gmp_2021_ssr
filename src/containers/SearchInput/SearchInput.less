@import "../../less/variables";

.searchInput {
	width: 100%;
	margin-right: 14px;
	padding: 0 20px;
	border: none;
	border-radius: 5px;
	color: @white;
	font-size: 22px;
	font-family: @NetflixSans-Regular;
	background-color: @dark-gray;
	letter-spacing: 0.7px;
	opacity: 0.8;
}
