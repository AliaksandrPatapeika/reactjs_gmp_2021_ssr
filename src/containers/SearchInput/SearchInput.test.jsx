import {noop} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import SearchInput from './SearchInput';

describe('SearchInput', () => {
  test('renders correctly', () => {
    const searchInput = renderer
      .create(<SearchInput value="value" onChange={noop} />)
      .toJSON();

    expect(searchInput).toMatchSnapshot();
  });
});
