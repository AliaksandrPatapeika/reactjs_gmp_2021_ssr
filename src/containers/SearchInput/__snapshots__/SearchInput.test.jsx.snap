// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<input
  autoComplete="off"
  className="searchInput"
  onChange={[Function]}
  placeholder="What do you want to watch?"
  value="value"
/>
`;
